### PR TITLE
Add spiral array in C

### PR DESCRIPTION
**Pull request by sandhyabhan**
<Issue number #960  >
Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added code for spiral array with test inputs.

- Added spaces around operator.

#### Languages Used:
- C
- 

#### Files Added:
- spiralarray.c
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raghav-dalmia** [comment]
don't create a new file. make changes in the existing one.

File: Spiral_Array/spiralarray.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
+include<stdio.h>
+int main()
+{
+   int a[20][20], int d,int i,int j,int ra,int ca,int r,int c,int k;
+    do
+    {
+       printf("enter number of rows:");         //accepting number of rows
+       scanf("%d" , &ra);
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Remove space before a comma. Also, add comments above the statement

File: Spiral_Array/spiralarray.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
+include<stdio.h>
+int main()
+{
+   int a[20][20], int d,int i,int j,int ra,int ca,int r,int c,int k;
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Add space after a comma.

File: Spiral_Array/spiralarray.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    do
+    {
+       printf("enter number of rows:");         //accepting number of rows
+       scanf("%d" , &ra);
+    }while(ra<1);
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Space around the operator.

File: Spiral_Array/spiralarray.c [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    do
+    {
+       printf("enter number of columns:");      //accepting number of columns
+       scanf("%d" , &ca);
+    }while(ca<1);
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Space around the operator.

File: Spiral_Array/spiralarray.c [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+   int a[20][20], int d,int i,int j,int ra,int ca,int r,int c,int k;
+    do
+    {
+       printf("enter number of rows:");         //accepting number of rows
+       scanf("%d" , &ra);
+    }while(ra<1);
+    do
+    {
+       printf("enter number of columns:");      //accepting number of columns
+       scanf("%d" , &ca);
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Remove space before a comma

File: Spiral_Array/spiralarray.c [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    }while(ca<1);
+    printf("\n\t enter elements:");
+    printf("\n");
+     for(i=0 ; i<ra ; i++)
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: add space around the operator

**Comment by somya-kapoor** [comment]
Please update the PR description, use [x] not [x ].
Also, update the file by adding space around the =, <, >, etc

**Comment by somya-kapoor** [comment]
Please update the PR description, use [x] not [x ].

File: Spiral_Array/spiralarray.c [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    do
+    {
+       printf("enter number of columns:");     
+       scanf("%d", &ca);
+    }while(ca < 1);
+    printf("\n\t enter elements:");
+    printf("\n");
+    //accepting matrix
+     for(i = 0 ; i < ra ; i++)
+    {
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Indentation.
Please remove space before;

File: Spiral_Array/spiralarray.c [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        for(j = 0 ; j < ca ; j++)
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Please remove space before ;

File: Spiral_Array/spiralarray.c [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            scanf("%d", &a[i][j]);      
+        }
+    }
+    printf(" INTIAL MATRIX = ");
+     for(i = 0 ; i < ra ; i++)
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Indentation for "for"

File: Spiral_Array/spiralarray.c [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        printf("\n");
+       
+        for(j = 0 ; j < ca ; j++)
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Please remove space before ;

File: Spiral_Array/spiralarray.c [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+            {
+                printf("%d", a[j][k]);                 
+            }
+           
+            j++;
+            d = 1;
+           
+        }
+       else if(d == 1)
+        {
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: indentation of {

File: Spiral_Array/spiralarray.c [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+            for(k = j ; k <= r ; k++)
+            {
+                printf("%d", a[k][c]);
+            }
+               
+            c--;
+            d = 2;
+        }
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: indentation of }

File: Spiral_Array/spiralarray.c [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            for(k = r ; k >= j ; k--)
+            {
+                printf("%d", a[k][i]);
+            }
+            i++;
+            d = 0;
+        }
+    }
+return 0;
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Indentation

**Comment by sandhyabhan** [comment]
I have made the changes..please check.

File: Spiral_Array/spiralarray.c (hunk 13 above)
Comment (somya-kapoor) [reply Spiral_Array/spiralarray.c]: Please correct the indentation of return

File: Spiral_Array/spiralarray.c [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+    printf("\n");
+    printf("spiral form = ");
+    printf("\n");
+    r = ra - 1;
+    c = ca - 1;
+    i = 0;
+    j = 0;
+    //printing each side so as to get  spiral form
+   while(i <= c && j <= r)
+    {
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Indentation of {

File: Spiral_Array/spiralarray.c (hunk 7 above)
Comment (somya-kapoor) [reply Spiral_Array/spiralarray.c]: Indentation of {

File: Spiral_Array/spiralarray.c [hunk 15]
```diff
@@ -0,0 +1,101 @@
+#include<stdio.h>
```
Comment (somya-kapoor) [inline Spiral_Array/spiralarray.c]: Please change the file to Spiral_Array/Spiral_Array.c

Comment (sandhyabhan) [reply Spiral_Array/spiralarray.c]: I tried doing so...but it is showing file name already exists...
what should i do?

File: Spiral_Array/SpiralArray.c [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+            }
+            i++;
+            d = 0;
+       }
+    }
+ return 0;
+}
+
+
+// Test Cases
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: use /* instead of //

File: Spiral_Array/SpiralArray.c [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+
+Input:
+        1    2   3   4
+        5    6   7   8
+        9   10  11  12
+        13  14  15  16
+Output: 
+1 2 3 4 8 12 16 15 14 13 9 5 6 7 11 10 
+
+//
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: use */ instead of using //

File: Spiral_Array/SpiralArray.c [hunk 18]
```diff
@@ -0,0 +1,101 @@
+#include<stdio.h>
+int main()
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: Leave a line before int main

File: Spiral_Array/SpiralArray.c [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+    //accepting number of columns
+    do
+    {
+       printf("enter number of columns:");     
+       scanf("%d", &ca);
+    }while(ca < 1);
+    printf("\n\t enter elements:");
+    printf("\n");
+    //accepting matrix
+     for(i = 0; i < ra; i++)
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+     {
+        for(j = 0; j < ca; j++)
+        {
+            scanf("%d", &a[i][j]);      
+        }
+    }
+    printf(" INTIAL MATRIX = ");
+     for(i = 0; i < ra; i++)
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+     }
+    printf("\n");
+    printf("spiral form = ");
+    printf("\n");
+    r = ra - 1;
+    c = ca - 1;
+    i = 0;
+    j = 0;
+    //printing each side so as to get  spiral form
+   while(i <= c && j <= r)
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            for(k = i; k <= c; k++)
+            {
+                printf("%d", a[j][k]);                 
+            }
+           
+            j++;
+            d = 1;
+           
+        }
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+       {
+            for(k = r; k >= j; k--)
+            {
+                printf("%d", a[k][i]);
+            }
+            i++;
+            d = 0;
+       }
+    }
+ return 0;
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+       {
+            for(k = r; k >= j; k--)
+            {
+               printf("%d", a[k][i]);
+            }
+            i++;
+            d = 0;
+       }
+    }
+ return 0;
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+}
+
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: remove this line

File: Spiral_Array/SpiralArray.c (hunk 21 above)
Comment (somya-kapoor) [reply Spiral_Array/SpiralArray.c]: inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 26]
```diff
@@ -0,0 +1,101 @@
+#include<stdio.h>
+
+int main()
+{
+   int a[20][20], int d, int i, int j, int ra, int ca, int r, int c, int k;
+   //accepting number of rows
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: inconsistency in indentation

File: Spiral_Array/SpiralArray.c (hunk 21 above)
Comment (somya-kapoor) [reply Spiral_Array/SpiralArray.c]: This change is not done

File: Spiral_Array/SpiralArray.c [hunk 27]
```diff
@@ -0,0 +1,99 @@
...
+       {
+            for(k = r; k >= j; k--)
+            {
+               printf("%d", a[k][i]);
+            }
+            i++;
+            d = 0;
+       }
+    }
+  return 0;
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: Inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 28]
```diff
@@ -0,0 +1,99 @@
...
+       else if(d == 1)
+       {
+            for(k = j; k <= r; k++)
+            {
+               printf("%d", a[k][c]);
+            }
+               
+            c--;
+            d = 2;
+        }
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: Inconsistency in indentation

File: Spiral_Array/SpiralArray.c (hunk 22 above)
Comment (somya-kapoor) [reply Spiral_Array/SpiralArray.c]: Inconsistency in indentation

File: Spiral_Array/SpiralArray.c [hunk 29]
```diff
@@ -0,0 +1,99 @@
...
+       }
+    }
+    printf("\n");
+    printf("spiral form = ");
+    printf("\n");
+    r = ra - 1;
+    c = ca - 1;
+    i = 0;
+    j = 0;
+   //printing each side so as to get  spiral form
```
Comment (somya-kapoor) [inline Spiral_Array/SpiralArray.c]: Inconsistency in indentation

**Comment by raghav-dalmia** [comment]
Looks good to me. Just squash your commits.

File: Spiral_Array/SpiralArray.c [hunk 30]
```diff
@@ -0,0 +1,97 @@
...
+       {
+            for(k = r; k >= j; k--)
+            {
+               printf("%d", a[k][i]);
+            }
+            i++;
+            d = 0;
+       }
+    }
+ return 0;
```
Comment (raghav-dalmia) [inline Spiral_Array/SpiralArray.c:85]: indentation issue

File: Spiral_Array/SpiralArray.c [hunk 31]
```diff
@@ -0,0 +1,97 @@
...
+       else if(d == 3)
+       {
+            for(k = r; k >= j; k--)
+            {
+               printf("%d", a[k][i]);
+            }
+            i++;
+            d = 0;
+       }
+    }
```
Comment (raghav-dalmia) [inline Spiral_Array/SpiralArray.c:84]: indentation issue

File: Spiral_Array/SpiralArray.c [hunk 32]
```diff
@@ -0,0 +1,97 @@
...
+       }
+       else if(d == 1)
+       {
+            for(k = j; k <= r; k++)
+            {
+               printf("%d", a[k][c]);
+            }
+            c--;
+            d = 2;
+        }
```
Comment (raghav-dalmia) [inline Spiral_Array/SpiralArray.c:65]: indentation issue

**Comment by raghav-dalmia** [comment]
Please squash your commits, rest looks good to me. I will merge it after that.

**Pull request merged**